lib/src/utils/path_utils.dart

```diff
@@ -111,3 +112,43 @@ Future<String> _getDefaultAndroidDir() async {
     }
   }
 }
+
+/// Searches for an empty filename in the given [saveDir] and returns it.
+/// If the file already exists, it will append a number to the filename in brackets.
+/// Returns only the name of the file as a string.
+Future<String> emptyFileName(String saveDir, String fileName, String? fileContentType) async {
+  String potentialFileName;
+  int counter = 0;
+  while (true) {
+    final baseWithoutExt = path.basenameWithoutExtension(fileName);
+
+    if (counter == 0) {
+      potentialFileName = baseWithoutExt;
+    } else {
+      potentialFileName = '$baseWithoutExt ($counter)';
+    }
+
+    var extension = path.extension(fileName); // includes '.'
+    if (extension.isNotEmpty) {
+      extension = extension.substring(1);
+    } else {
+      extension = mime.extensionFromMime(fileContentType ?? ''); // excludes '.'
```

Comment:
Wow nice catch, that is some weird behaviour 👀
This should work around it? https://github.com/ar-io/ardrive_io/pull/10/commits/46f2c1161b943c618fad572e58c168ff398f01da

Reply:
Yeah... Strange behavior. That should do the trick for most cases, yes!
I'd say it would only fail (i.e. `fileExtensionOrMime != fileContentType` will be true while both are the mime) for non-lower-case mimes (such as when the user sets a custom Content-Type) because of the first line of the `extensionFromMime` method... We may work around that by either making `fileContentType` to be the lower-case version of it or by calling `toLowerCase` on `fileContentType` in the `if` you added there.

Reply:
Ahh that's why you read the code not the docs! My bad!